web/server/codechecker_server/api/report_server.py

```diff
@@ -3279,3 +3279,124 @@ def getAnalysisStatistics(self, run_id, run_history_id):
+
+            # Logic for getting comments
+            comment_data_list = defaultdict(list)
+            comment_query = session.query(Comment, Report.bug_id) \
+                .filter(Report.run_id.in_(run_ids)) \
+                .outerjoin(Report, Report.bug_id == Comment.bug_hash) \
+                .order_by(Comment.created_at.desc())
+
+            for data, report_id in comment_query:
+                comment_data = comment_data_db_to_api(data)
+                comment_data_list[report_id].append(comment_data)
+
+            # Logic for getting review status
+            review_data_list = {}
+            review_query = session.query(ReviewStatus, Report.bug_id) \
+                .filter(Report.run_id.in_(run_ids)) \
+                .outerjoin(Report, Report.bug_id == ReviewStatus.bug_hash) \
+                .order_by(ReviewStatus.date.desc())
+
+            for data, report_id in review_query:
+                review_data = create_review_data(data)
+                review_data_list[report_id] = review_data
+
+        return ExportData(comments=comment_data_list,
+                          reviewData=review_data_list)
+
+    @exc_to_thrift_reqfail
+    @timeit
+    def importData(self, exportData):
+        self.__require_admin()
+        with DBSession(self.__Session) as session:
+
+            # Logic for importing comments
+            imported_data = {}
+            imported_comments = []
+
+            for bug_hash, comments in exportData.comments.items():
+                for comment in comments:
+                    imported_data[comment.id] = [bug_hash, comment]
+                    imported_comments.append(comment)
+
+            comment_bug_ids = list(exportData.comments.keys())
+            current_query = session.query(Comment)\
+                .filter(Report.bug_id.in_(comment_bug_ids)) \
+                .outerjoin(Report, Report.bug_id == Comment.bug_hash) \
+                .order_by(Comment.created_at.desc())
+
+            db_comments = {}
+            for comment in current_query:
+                db_comments[comment.id] = comment
+
+            for imported_comment in imported_comments:
+                # Check if the new comment exists in the database
+                if imported_comment.id in db_comments.keys():
+
+                    db_comment = db_comments.get(imported_comment.id)
+                    message = get_comment_msg(db_comment)
+
+                    # Check if the data is same for both
+                    if imported_comment.message != message:
+                        db_comment.message = imported_comment.message.encode(
+                            'utf-8')
+                        session.add(db_comment)
+
+                # If the comment does not exist add it
+                else:
+                    data = imported_data.get(imported_comment.id)
+                    new_comment = self.__add_comment(data[0], data[1].message)
+                    session.add(new_comment)
```

Comment:
Importing comments to the database is not so easy.
For example we can't use the comment id. Lets suppose that in the 2 users are using different servers. Both user insert a new comment on their own server with different messages. Both comment will have ID-1. If the first user exports the data from the server and send it to the second user who tries to import it, he would like to see both comments. So we can't say if the comment ID's are equal we will just update the comment.
To see if a comment is in the database we can check if the date, kind and messages fields are equals. If not, we will insert a new comment.

What do you think?

```suggestion
                        comment_bug_ids = list(exportData.comments.keys())
            comment_query = session.query(Comment) \
                .filter(Comment.bug_hash.in_(comment_bug_ids)) \
                .order_by(Comment.created_at.desc())

            comments_in_db = defaultdict(list)
            for comment in comment_query:
                comments_in_db[comment.bug_hash].append(comment)

            for bug_hash, comments in exportData.comments.items():
                db_comments = comments_in_db[bug_hash]
                for comment in comments:
                    date = datetime.strptime(comment.createdAt,
                                             '%Y-%m-%d %H:%M:%S.%f')

                    message = comment.message.encode('utf-8') \
                        if comment.message else b''

                    # See if the comment is already in the database.
                    if any(c.created_at == date and \
                           c.kind == comment.kind and \
                           c.message == message for c in db_comments):
                        continue

                    c = Comment(bug_hash, comment.author, message,
                                comment.kind, date)
                    session.add(c)
```

Reply:
Yes, this is certainly a better approach over just comparing `id` and `message`. I had followed the suggestion given by you in this comment thread: https://github.com/Ericsson/codechecker/issues/3076#issuecomment-747300763